services/cloudsearchdomain/src/main/java/software/amazon/awssdk/services/cloudsearchdomain/internal/SwitchToPostInterceptor.java

```diff
@@ -43,6 +44,8 @@ public SdkHttpRequest modifyHttpRequest(Context.ModifyHttpRequest context, Execu
             return httpRequest.toBuilder()
                               .clearQueryParameters()
                               .method(SdkHttpMethod.POST)
+                              .putHeader("Content-Type", singletonList("application/x-www-form-urlencoded; charset="
```

Comment:
Should we null check first?

Reply:
As discussed offline, JsonProtocolFactory automatically sets the value to `application/x-amz-json-1.1` which is causing the api to fail. So this value will not be null.